source/geod24/concurrency.d

```diff
@@ -1846,55 +1802,138 @@ private
      * A MessageBox is a message queue for one thread.  Other threads may send
      * messages to this owner by calling put(), and the owner receives them by
      * calling get().  The put() call is therefore effectively shared and the
-     * get() call is effectively local.  setMaxMsgs may be used by any thread
-     * to limit the size of the message queue.
+     * get() call is effectively local.
      */
     class MessageBox
     {
-        this() @trusted nothrow /* TODO: make @safe after relevant druntime PR gets merged */
+        this () @trusted nothrow /* TODO: make @safe after relevant druntime PR gets merged */
         {
-            m_lock = new Mutex;
-            m_closed = false;
+            this.mutex = new Mutex();
+            this.qsize = 64;
```

Comment:
The message box limit is hardcoded to 64 messages now? What happens when the limit is reached? Does the queue block the thread on read/write, or throws an exception, or something else happens?

Reply:
There are three kinds of queues.
One is the sending fiber and the queue that stores the context, the other is the queue that stores only the data, and the other is the  receiving fiber and the context.
qsize here is the second queue size.
If the queue fills up second, first queue or the third saved in the queue.
Hardcoding the size of the queue here is problematic. Let's think about another way.
Let's think about the way to remove the second queue